### PR TITLE
[functions] add if function

### LINKED ISSUE #918: [functions] Conditional Functions
**Summary**

If
Controls conditional branching. Unlike most systems, ClickHouse always evaluate both expressions then and else.
Syntax:
```
SELECT if(cond, then, else)
```

Example:

Query:
```
SELECT if(1, 2+2, 2+6);
```
Result:
```
┌─plus(2, 2)─┐
│          4 │
└────────────┘
```

Query:
```
SELECT if(0, 2+2, 2+6);
```

Result:
```
┌─plus(2, 6)─┐
│          8 │
└────────────┘
```

Reference:
https://clickhouse.tech/docs/en/sql-reference/functions/conditional-functions/#if

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

```
SELECT if(cond, then, else)
```

## Changelog

- New Feature

## Related Issues

Fixes #918 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/datavalues/src/arrays/ops/if.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+use common_arrow::arrow::array::ArrayRef;
+use common_arrow::arrow::compute::if_then_else;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::prelude::*;
+
+macro_rules! impl_if_common {
+    ($lhs:ident, $rhs:ident,$predicate:ident) => {{
+        match ($predicate.len(), $lhs.len(), $rhs.len()) {
```
Comment (sundy-li) [inline common/datavalues/src/arrays/ops/if.rs:26]: Why not keep the order as macros define.

File: common/datavalues/src/columns/conditional.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        }
+        let mut right = right.clone();
+        if right.data_type() != dtype {
+            right = right.cast_with_type(&dtype)?;
+        }
+
+        let len = self.len();
+        match left.data_type() {
+            DataType::Null => Ok(lhs.resize_constant(lhs.len())),
+            DataType::Boolean => try_get_if_result! {left, right, cond, bool, len},
```
Comment (sundy-li) [inline common/datavalues/src/columns/conditional.rs:54]: Can we add a method in SeriesTrait, then there will be no need to have this macros.

**Comment by zhyass** [comment]
TODO: 
Unit Tests
Stateless Tests

File: common/datavalues/src/columns/conditional.rs (hunk 2 above)
Comment (zhyass) [reply common/datavalues/src/columns/conditional.rs:54]: Sure, I will do it

File: common/datavalues/src/arrays/ops/if.rs (hunk 1 above)
Comment (zhyass) [reply common/datavalues/src/arrays/ops/if.rs:26]: fix

**Review by sundy-li — APPROVED** [review]
LGTM

**Pull request merged**